src/main/java/redis/clients/jedis/BinaryJedis.java

```diff
@@ -2209,7 +2203,11 @@ public Long zcount(final byte[] key, final byte[] min, final byte[] max) {
   private Set<Tuple> getBinaryTupledSet() {
     checkIsInMulti();
     List<byte[]> membersWithScores = client.getBinaryMultiBulkReply();
-    Set<Tuple> set = new LinkedHashSet<Tuple>();
+    if (membersWithScores.size() == 0) {
+      return Collections.emptySet();
+    }
+    Set<Tuple> set = new LinkedHashSet<Tuple>(
+        CollectionUtils.mapCapacity(membersWithScores.size() / 2));
```

Comment:
@itugs I don't totally understand this. Why do we need to calculate an "optimal" capacity depending on the growth if the resulting map will never grow?. We already have the final map size from the `membersWithScores` variable, so why just instantiating a `LinkedHashSet<Tuple>(membersWithScores.size())` isn't enough?. We know for sure that the map will be that size and it will not grow.